src/thor/costmatrix.cc

```diff
@@ -896,29 +904,36 @@ void CostMatrix::CheckReverseConnections(const uint32_t target,
         uint32_t d = std::abs(static_cast<int>(rev_pred.path_distance()) +
                               static_cast<int>(fwd_label.path_distance()) -
                               static_cast<int>(fwd_label.transition_cost().secs));
-        best_connection_[source_idx].Update(fwd_edgeid, rev_pred.edgeid(), Cost(s, s), d);
-        best_connection_[source_idx].found = true;
+        best_connection_[idx].Update(fwd_edgeid, rev_pred.edgeid(), Cost(s, s), d);
+        best_connection_[idx].found = true;
+
+        // Update status and update threshold if this is the last location
+        // to find for this source or target
+        UpdateStatus(source, target);
+      } else if (fwd_predidx == kInvalidLabel) {
```

Comment:
this is the actual bug fix. before we'd go directly into the `else` below in case it wasn't a trivial route. but that block of code looks at the opposing's `transition_cost`, which is not what we'd expect if the opposing is a correlated edge: in `SetTargets` we use `transition_cost` to store the full edge's length and the full duration for it. 

so this needs to be handled in a different way to get the connection's length & duration